### ISSUE TITLE
Execute through command?

### ISSUE THREAD

**Issue by Sinasta** [brunopostle/homemaker-addon#3]
I would like to ask if/how it is possible to topologoise a selscted object with a script.

![Screenshot from 2021-03-31 17-31-20](https://user-images.githubusercontent.com/64699866/113170545-f0305f00-9246-11eb-8805-f96e839ce9b5.png)

I would like to integrate this addon into my sverchok tree so that the addon gets automatically axecuted on a selected object. Thanks for your help!

**Comment by brunopostle** [comment]
The Homemaker addon is very scriptable, just call `bpy.ops.object.homemaker()` on any object, though I'm not sure how you get sverchok to create a 'normal' blender object, maybe you have to bake it.

Internally, the addon is just processing a list of nodes and faces, so it ought to be possible to write a sverchok node that does the same thing with sverchok data

**Comment by Sinasta** [comment]
thank you for your answer. i am new to blender scripting so i am quiet sure that i am doing something wrong here but when i run
 
bpy.ops.object.homemaker()

with an object selected i get this error:

Error: Python: Traceback (most recent call last):
  File "/home/sinasta/.config/blender/2.93/scripts/addons/homemaker-addon-main/__init__.py", line 101, in execute
    bpy.ops.object.hide_view_set(unselected=False)
  File "/usr/share/blender/2.93/scripts/modules/bpy/ops.py", line 132, in __call__
    ret = _op_call(self.idname_py(), None, kw)
RuntimeError: Operator bpy.ops.object.hide_view_set.poll() failed, context is incorrect

location: /usr/share/blender/2.93/scripts/modules/bpy/ops.py:132


Traceback (most recent call last):
  File "/usr/lib/python3.9/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<blender_console>", line 1, in <module>
  File "/usr/share/blender/2.93/scripts/modules/bpy/ops.py", line 132, in __call__
    ret = _op_call(self.idname_py(), None, kw)
RuntimeError: Error: Python: Traceback (most recent call last):
  File "/home/sinasta/.config/blender/2.93/scripts/addons/homemaker-addon-main/__init__.py", line 101, in execute
    bpy.ops.object.hide_view_set(unselected=False)
  File "/usr/share/blender/2.93/scripts/modules/bpy/ops.py", line 132, in __call__
    ret = _op_call(self.idname_py(), None, kw)
RuntimeError: Operator bpy.ops.object.hide_view_set.poll() failed, context is incorrect

location: /usr/share/blender/2.93/scripts/modules/bpy/ops.py:132

Am i missing a step? Again, the error is defenitly on my side so this is not an Issue with your script at all. Thanks for your incredible work!

**Comment by brunopostle** [comment]
It works if you comment-out line 101 in `__init__.py`, i.e. this line:

    #bpy.ops.object.hide_view_set(unselected=False)

I'm not sure why, all it does is hide the original mesh so the new geometry appears to replace it.

**Comment by Sinasta** [comment]
Thank you so much! works perfectly now.

**Comment by brunopostle** [comment]
This should be fixed in git, the original 'sketch' geometry still gets hidden, but you can call the add-on using `bpy.ops.object.homemaker()` without an error.

**Comment by Sinasta** [comment]
thats awesome! i will try it out later. i have another question. Do you plan to make the addon work with multiple objects selected? And what is your opinion on integrating the addon into sverchok?

**Comment by brunopostle** [comment]
Yes it would be good to be able to process multiple objects as separate buildings, this would be just another loop.

How do you think it would work with sverchok? There could be a node that takes a geometry and processes into a building, but this could be done with a SN node. Something that would be fun would be to set the style/material by distance above ground or by distance from the entrance.

**Comment by Sinasta** [comment]
i managed to make the addon work on multiple objects. I just setup a simple loop to select one after enother and than executed bpy.ops.object.homemaker() for each. So it should probably be not to hard to implement that directly in the addon. The integration with sverchok would certainly be super helpful in many different ways. I don't know enough about coding to propose a functinality but simply the fact to make the addon work parametrically would be great. Do you think that there could be a possible integration with the pyclone library? https://github.com/CreativeDesigner3D/PyClone and https://github.com/CreativeDesigner3D/Library_Home_Builder this could possibly able the user to tweak the different elements (windows, doors) to their liking

**Comment by brunopostle** [comment]
I added a loop that should process multiple meshes as multiple buildings.

Assets like windows and doors need a more usable interface. Currently they are defined as DXF meshes and read by a Perl library (File::DXF). As I port everything to python I will temporarily switch to ezdxf and read the same meshes. Then I'm not sure, they could all be stored as mapped IFC objects in an IFC file, or using the new blender asset manager, or pyclone. I want the addon to work in freecad as well as blender, so this points to an IFC file, but this would need a nice interface for editing. I won't make any decisions until I have investigated both the blender asset manager and pyclone.
